### ISSUE TITLE
Extending the Model class

### ISSUE THREAD

**Issue by JuhG** [robindrost/prismic-eloquent#3]
Hey, cool idea!
Is there any reason that you're not actually extending the Eloquent Model class? Could it be possible to only change the connection to use the API instead of the database? This ways we could leave the rest as is to lose as little from the existing functionality as possible.

**Comment by robindrost** [comment]
Unfortunately this is not possible because of the big differences between a database and the Prismic api. For example relatationships, updating, deleting etc. work totally different or not at all. The idea is to make retrieving data from Prismic as close as possible to the Eloquent way.

**Comment by JuhG** [comment]
I have to disagree. I've put together a proof-of-concept of my idea in this repo:
https://github.com/dromedar-design/laravel-prismic
This is definitely not as clean as your solution and there are a lot of unused functionality, but you still get real nice things out of the box extending the official classes.
Also I've added a few templating shortcuts and plan to finetune it as I work on a client project.

**Comment by robindrost** [comment]
Wow cool, looks really nice! 

I understand it makes it feel more "real eloquent" but I still think the real Laravel model / eloquent situation serves a different purpose. The functionality I've build just makes it feel a little more like eloquent but its def. not the same. Beside that you will need to overwrite a lot you don't really need from extending the class.

Do relationships etc. already work in the repo? For now I have oneToMany and ManyToMany and realised that for example belongsTo is not gonna work with Prismic. Any ideas?

Would like to hear more if you go further with this project. Love the effort!

**Comment by JuhG** [comment]
Thanks for the kind words!

I absolutely agree, that it _feels_ like a workaround, because I don't use a lot of things and overwrite stuff completely differently. And it does feel wrong. But...

First of all I've learned a lot while digging deep into the source code and it's awesome knowing how things work behind the scenes. (It has nothing to do with Prismic, just wanted to mention it :) 

In the end I still think that this is the right way and it's actually Laravel's fault, that we feel it's wrong in the sense, that the Query\Builder class is really tightly coupled to SQL databases to the point that it all leads to the toSql() method, where the whole querying process boils down to a single string. Which is in many cases the wrong approach. The querying process should result in an array (or similar), that could be translated into the data sources format - I'm doing something similar, using the Builder object as a "config object", that holds all the info and transforming (with the Statement class) into something usable (in our case into an array of predicates). I think the original purpose of Grammar is exactly that, I might revisit that, but it seemed too coupled with SQL and creating one string in the end.

So I might be completely wrong, or maybe we see changes in Laravel 6, who knows. I'll definitely ask in the Laravel repo if this is indeed a somewhat correct path or it's completely against the original purpose.

In its current form I still like that most things work out of the box, like using primaryKey and table variables is cool. All the built in where... methods work and also relationships, if you name them correctly - category_uid by default, since I use uid as a primary key.

I still need to correct a few things (OR queries don't work, they need to make multiple api calls) and I'll work on the templating part as I work on a project, but all in all I like the direction.

Also I took quite some inspiration from your package, hope you don't mind.
